test/inference/gibbs_conditional.jl

```diff
@@ -0,0 +1,122 @@
+using Random, Turing, Test
+using Clustering
+
+dir = splitdir(splitdir(pathof(Turing))[1])[1]
+include(dir*"/test/test_utils/AllUtils.jl")
+
+
+@turing_testset "gibbs conditionals" begin
+    let N = 1000,
+        (α_0, θ_0) = (2.0, inv(3.0)),
+        λ_true = rand(Gamma(α_0, θ_0)),
+        σ_true = sqrt(1 / λ_true),
+        m_true = rand(Normal(0, σ_true)),
+        x = rand(Normal(m_true, σ_true), N)
+        
+        @model function inverse_gdemo(x)
+            λ ~ Gamma(α_0, θ_0)
+            m ~ Normal(0, sqrt(1 / λ))
+            @. x ~ $(Normal(m, sqrt(1 / λ)))
+        end
+
+        function gdemo_statistics(x)
+            # The conditionals and posterior can be formulated in terms of the following statistics:
+            n = length(x) # number of samples
+            x_bar = mean(x) # sample mean
+            s2 = var(x; mean=x_bar, corrected=false) # sample variance
+            return n, x_bar, s2
+        end
+
+        function cond_m(c)
+            n, x_bar, s2 = gdemo_statistics(x)
+            m_n = N * x_bar / (n + 1)
+            λ_n = c.λ * (n + 1)
+            σ_n = sqrt(1 / λ_n)
+            return Normal(m_n, σ_n)
+        end
+
+        function cond_λ(c)
+            n, x_bar, s2 = gdemo_statistics(x)
+            α_n = α_0 + (n - 1) / 2 + 1
+            β_n = s2 * n / 2 + c.m^2 / 2 + inv(θ_0)
+            return Gamma(α_n, inv(β_n))
+        end
+
+        # Three tests: one for each variable fixed to the true value, and one for both
+        # using the conditional
+        Random.seed!(100)
+        for alg in (Gibbs(GibbsConditional(:m, cond_m),
+                          GibbsConditional(:λ, c -> Normal(λ_true, 0))),
+                    Gibbs(GibbsConditional(:m, c -> Normal(m_true, 0)),
+                          GibbsConditional(:λ, cond_λ)),
+                    Gibbs(GibbsConditional(:m, cond_m),
+                          GibbsConditional(:λ, cond_λ)))
+            chain = sample(inverse_gdemo(x), alg, 10_000)
+            check_numerical(chain, [:m, :λ], [m_true, λ_true], atol=0.2)
+        end
+    end
+
+    Random.seed!(100)
+    let π = [0.5, 0.5],
+        K = length(π),
+        m = 0.5,
+        λ = 2.0,
```

Comment:
Again I'm curious why you use a `let` block here. It seems also a bit inconsistent/confusing to use `λ` for the precision above and for the standard deviation here.